### PR TITLE
perl-constant: add new package

### PR DESCRIPTION
**Pull request by ChristopherChristofi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/perl-constant/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/perl-constant/package.py]:
```suggestion
# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
```

Update license for 2024.

**Review by alecbcs — CHANGES_REQUESTED** [review]
Confirmed SHA. PR looks good to me. Just a quick request to update the license for 2024.

**Review by alecbcs — APPROVED** [review]
Confirmed SHA. Looks good to me.

**Comment by ChristopherChristofi** [comment]
```
$ corelist constant

Data for 2023-07-02
constant was first released with perl 5.004
```
Has been a core module since version 5.004, which pre-dates all end of life Perl releases available in spack, and therefore already available with each spack installed version of Perl

**Pull request merged**